### PR TITLE
[EventKit] Add support for Xcode 15 beta 7

### PR DESCRIPTION
**Pull request by haritha-mohan**

### CODE REVIEW

File: src/eventkit.cs [hunk 1]
```diff
@@ -672,16 +672,36 @@ interface EKEventStore {
...
+		[Deprecated (PlatformName.WatchOS, 10, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
+		[Deprecated (PlatformName.MacCatalyst, 17, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
 		[Export ("requestAccessToEntityType:completion:")]
 		[Async]
 		void RequestAccess (EKEntityType entityType, Action<bool, NSError> completionHandler);
 
+		[Watch (10, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("requestFullAccessToEventsWithCompletion:")]
+		[Async]
+		void RequestFullAccessToEventsWithCompletion (EKEventStoreRequestAccessCompletionHandler completion);
```
Comment (dalexsoto) [inline src/eventkit.cs:686]:
```suggestion
		void RequestFullAccessToEvents (EKEventStoreRequestAccessCompletionHandler completion);
```

File: src/eventkit.cs [hunk 2]
```diff
@@ -672,16 +672,36 @@ interface EKEventStore {
...
+
+		[Watch (10, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("requestWriteOnlyAccessToEventsWithCompletion:")]
+		[Async]
+		void RequestWriteOnlyAccessToEventsWithCompletion (EKEventStoreRequestAccessCompletionHandler completion);
```
Comment (dalexsoto) [inline src/eventkit.cs:691]:
```suggestion
		void RequestWriteOnlyAccessToEvents (EKEventStoreRequestAccessCompletionHandler completion);
```

File: src/eventkit.cs [hunk 3]
```diff
@@ -672,16 +672,36 @@ interface EKEventStore {
 		EKSource [] DelegateSources { get; }
 
 		[MacCatalyst (13, 1)]
+		[Deprecated (PlatformName.iOS, 17, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
+		[Deprecated (PlatformName.MacOSX, 14, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
+		[Deprecated (PlatformName.WatchOS, 10, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
+		[Deprecated (PlatformName.MacCatalyst, 17, 0, message: "Use RequestFullAccessToEventsWithCompletion, RequestWriteOnlyAccessToEventsWithCompletion, or RequestFullAccessToRemindersWithCompletion.")]
```
Comment (dalexsoto) [inline src/eventkit.cs:678]:
```suggestion
		[Deprecated (PlatformName.iOS, 17, 0, message: "Use RequestFullAccessToEvents, RequestWriteOnlyAccessToEvents, or RequestFullAccessToReminders.")]
		[Deprecated (PlatformName.MacOSX, 14, 0, message: "Use RequestFullAccessToEvents, RequestWriteOnlyAccessToEvents, or RequestFullAccessToReminders.")]
		[Deprecated (PlatformName.WatchOS, 10, 0, message: "Use RequestFullAccessToEvents, RequestWriteOnlyAccessToEvents, or RequestFullAccessToReminders.")]
		[Deprecated (PlatformName.MacCatalyst, 17, 0, message: "Use RequestFullAccessToEvents, RequestWriteOnlyAccessToEvents, or RequestFullAccessToReminders.")]
```

File: src/eventkit.cs [hunk 4]
```diff
@@ -672,16 +672,36 @@ interface EKEventStore {
...
+
+		[Watch (10, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("requestWriteOnlyAccessToEventsWithCompletion:")]
+		[Async]
+		void RequestWriteOnlyAccessToEventsWithCompletion (EKEventStoreRequestAccessCompletionHandler completion);
+
+		[Watch (10, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("requestFullAccessToRemindersWithCompletion:")]
+		[Async]
+		void RequestFullAccessToRemindersWithCompletion (EKEventStoreRequestAccessCompletionHandler completion);
```
Comment (dalexsoto) [inline src/eventkit.cs:696]:
```suggestion
		void RequestFullAccessToReminders (EKEventStoreRequestAccessCompletionHandler completion);
```

**Review by dalexsoto — CHANGES_REQUESTED** [review]
Awesome job!

**Review by rolfbjarne — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]
Assuming all comments from @dalexsoto  are addressed

**Review by tj-devel709 — APPROVED** [review]

**Comment by haritha-mohan** [comment]
Test failures are unrelated (https://github.com/xamarin/maccore/issues/2691).

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**